resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreakerConfig.java

```diff
@@ -248,7 +250,9 @@ public CircuitBreakerConfig build() {
             config.failureRateThreshold = failureRateThreshold;
             config.ringBufferSizeInClosedState = ringBufferSizeInClosedState;
             config.ringBufferSizeInHalfOpenState = ringBufferSizeInHalfOpenState;
-            config.recordFailurePredicate = errorRecordingPredicate;
+            if(errorRecordingPredicate != null) {
```

Comment:
This is the only functional change in this PR. Previously a null `errorRecordingPredicate` would allow being used, but that seem undesired from what I could see in the code.